src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Components/TagHelperDescriptorExtensions.cs

```diff
@@ -46,6 +48,53 @@ public static bool IsGenericTypedComponent(this TagHelperDescriptor tagHelper)
             string.Equals(bool.TrueString, value);
     }
 
+    /// <summary>
+    /// Given a taghelper binding it finds the BoundAttribute that is a type parameter and then the
+    /// actual binding value for that type.
+    ///
+    /// <code>
+    /// &lt;MyTagHelper
+    ///   TItem="string"
+    ///   OnChange="OnMyTagHelperChange" /&gt;
+    /// </code>
+    ///
+    /// The above code will return "string" for the typeName.
+    /// </summary>
+    /// <param name="tagHelper"></param>
+    /// <param name="binding"></param>
+    /// <param name="typeName"></param>
+    /// <returns></returns>
+    public static bool TryGetGenericTypeName(this TagHelperDescriptor tagHelper, TagHelperBinding binding, [NotNullWhen(true)] out string typeName)
+    {
+        if (!tagHelper.IsComponentTagHelper)
```

Comment:
It looks like the method only works for components, not normal "tag helpers", consider renaming it and updating its doc string to make that clearer.